Jenkinsfile

```diff
@@ -1,14 +1,69 @@
+def nexusSnapshotRepoDetails = getNexusSnapshotRepoDetails()
+def nexusReleaseRepoDetails = getNexusReleaseRepoDetails()
+def mavenSettingsId = getMavenSettingsId()
+def mavenAdditionalSettings = getMavenAdditionalSettings()
+def enceladusSlaveLabel = getEnceladusSlaveLabel()
+def deploymentStageRun = 'false'
+
 pipeline {
-     agent {
-         docker {
-             image 'maven:3-alpine'
-         }
-     }
-     stages {
-         stage('Build') {
-             steps {
-                 sh 'mvn -B -DskipTests clean install'
-             }
-         }
-     }
- }
+    agent {
+        label "${enceladusSlaveLabel}"
+    }
+    tools {
+        jdk 'openjdk-1.8.0'
+        maven 'Maven-3.6.0'
+        git 'git-latest'
```

Comment:
I would maybe parametrize the tools as well - but that might be too much

Reply:
This I can indeed abstract further.

Reply:
Tool versions abstracted, resolving.